### PR TITLE
separate azbi from e structures

### PR DESCRIPTION
**Pull request by mkyc**
this PR removes dependencies to e-structures library, moves all related tests into this repository and also refactors some of module mechanisms.

### CODE REVIEW

File: internal/input/input.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		return err
+	}
+	bytes, err := json.MarshalIndent(i, "", "\t")
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(path, bytes, 0644)
+}
+
+func BackupRaw(existingPath, newPath string) error {
```
Comment (erzetpe) [inline internal/input/input.go:143]: Where is this BackupRaw used?

File: internal/input/input.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+				}
+				if !found {
+					sl.ReportError(
+						params.VmGroups[i].SubnetNames[j],
+						fmt.Sprintf("VmGroups[%d].SubnetNames[%d]", i, j),
+						fmt.Sprintf("SubnetNames[%d]", j),
+						"insubnets",
+						"")
+					return
+				}
```
Comment (erzetpe) [inline internal/input/input.go:193]: This code would be better to extract somewhere it doesn't look good.

**Review by erzetpe — CHANGES_REQUESTED** [review]

File: internal/terraformer/terraformer.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
+package terraformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/epiphany-platform/m-azure-basic-infrastructure/internal/input"
+	"github.com/epiphany-platform/m-azure-basic-infrastructure/internal/log"
+	terra "github.com/mkyc/go-terraform"
```
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/terraformer/terraformer.go:8]: I do not know TBH. Code in there is so ugly that I'm not sure 🙈

File: internal/output/output.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+	return output, output.validate()
+}
+
+func Backup(o *Output, path string) error {
+	if _, err := os.Stat(path); err == nil {
+		// file does exist
+		return os.ErrExist
+	} else if os.IsNotExist(err) {
+		// ok, file does not exist
```
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/output/output.go:100]: This code returns errors correctly and comments are just for developers to understand how to correctly handle file existence check.

File: internal/input/input.go (hunk 1 above)
Comment (mkyc) [reply internal/input/input.go:143]: In `processor` package it each `Pre` function.

File: internal/input/input.go (hunk 2 above)
Comment (mkyc) [reply internal/input/input.go:193]: What is wrong with it? Its just simple `for`s and only long here is `sl.ReportError()` function which has multiple parameters and that is it.

File: internal/to/to.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package to
```
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/to/to.go]: Good point. Changed.

File: internal/utils/test/errors.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	"testing"
+)
+
+type ValidationErrors []ValidationError
+
+func (e ValidationErrors) Error() string {
+	buff := bytes.NewBufferString("")
+
+	for _, te := range e {
+
```
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/utils/test/errors.go]: Fixed.

File: internal/terraformer/terraformer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/terraformer/terraformer.go:8]: I will, but not today :P

**Review by erzetpe — CHANGES_REQUESTED** [review]

File: internal/output/output.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkyc) [reply internal/output/output.go]: Comments in this section fixed.

[the remark this answers is not in the document]
Comment (erzetpe) [reply internal/output/output.go]: Good. :)

File: internal/input/input.go (hunk 2 above)
Comment (erzetpe) [reply internal/input/input.go:193]: I am not a big fan of it. But it can stay as it is.

**Review by erzetpe — APPROVED** [review]

**Pull request merged**